### PR TITLE
memory manager: Add runtime memory discovery

### PR DESCRIPTION
**Pull request by dabekjakub**
Add runtime physical memory discovery.
This allows platforms with different physical memory sizes to use the same driver for memory management. 

Signed-off-by: Jakub Dabek <jakub.dabek@intel.com>

### CODE REVIEW

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c [hunk 1]
```diff
@@ -642,6 +642,13 @@ static int sys_mm_drv_mm_init(const struct device *dev)
 
 	ARG_UNUSED(dev);
 
+	/*
+	 * Change size of avalible physical memory acording to fw register information
+	 * in runtime.
+	 */
+	uint32_t avalible_memory_size = mtl_hpsram_get_bank_count() * SRAM_BANK_SIZE;
+	L2_PHYS_SRAM_REGION.num_blocks = avalible_memory_size/CONFIG_MM_DRV_PAGE_SIZE;
```
Comment (dcpleung) [inline drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:650]: Need spaces around `/`

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c [hunk 2]
```diff
@@ -642,6 +642,13 @@ static int sys_mm_drv_mm_init(const struct device *dev)
 
 	ARG_UNUSED(dev);
 
+	/*
+	 * Change size of avalible physical memory acording to fw register information
+	 * in runtime.
+	 */
+	uint32_t avalible_memory_size = mtl_hpsram_get_bank_count() * SRAM_BANK_SIZE;
```
Comment (nashif) [inline drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:649]: I am sure this does not exist in upstream zephyr, try `ace_hpsram_get_bank_count`

**Review by nashif — CHANGES_REQUESTED** [review]

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c (hunk 2 above)
Comment (dabekjakub) [reply drivers/mm/mm_drv_intel_adsp_mtl_tlb.c]: Was working with bad revision - fixed.

File: drivers/mm/mm_drv_intel_adsp_mtl_tlb.c (hunk 1 above)
Comment (dcpleung) [reply drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:652]: Still need spaces.

Comment (dabekjakub) [reply drivers/mm/mm_drv_intel_adsp_mtl_tlb.c:652]: fixed - sorry misunderstood.

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**